README.md

```diff
@@ -4,23 +4,66 @@
 
 ## Table of Contents
 
-- [About](#about)
-- [Design](#design)
-- [Prerequisites](#prerequisites)
-- [Supported Kubernetes versions](#supported-kubernetes-versions)
-- [Setup](#setup)
-- [Automated testing](#automated-testing)
-- [Communication and contribution](#communication-and-contribution)
+- [Intel PMEM-CSI plugin for Kubernetes](#intel-pmem-csi-plugin-for-kubernetes)
+    - [Table of Contents](#table-of-contents)
+    - [About](#about)
+    - [Design](#design)
+        - [Architecture and Operation](#architecture-and-operation)
+        - [DeviceMode:LVM](#devicemodelvm)
+            - [Namespace modes in DeviceMode:LVM](#namespace-modes-in-devicemodelvm)
+            - [Using limited amount of total space in DeviceMode:LVM](#using-limited-amount-of-total-space-in-devicemodelvm)
+        - [DeviceMode:Direct](#devicemodedirect)
+            - [Namespace modes in DeviceMode:Direct](#namespace-modes-in-devicemodedirect)
+            - [Using limited amount of total space in DeviceMode:Direct](#using-limited-amount-of-total-space-in-devicemodedirect)
+        - [Driver modes](#driver-modes)
+        - [Driver Components](#driver-components)
+            - [Identity Server](#identity-server)
+            - [Node Registry Server](#node-registry-server)
+            - [Master Controller Server](#master-controller-server)
+            - [Node Controller Server](#node-controller-server)
+            - [Node Server](#node-server)
+        - [Communication channels](#communication-channels)
+        - [Security](#security)
+        - [Volume Persistency](#volume-persistency)
+            - [Usage on Kubernetes](#usage-on-kubernetes)
+    - [Prerequisites](#prerequisites)
+        - [Software required](#software-required)
+        - [Hardware required](#hardware-required)
+        - [Persistent memory device initialization](#persistent-memory-device-initialization)
+    - [Supported Kubernetes versions](#supported-kubernetes-versions)
+    - [Setup](#setup)
+        - [Get source code](#get-source-code)
+        - [Build plugin](#build-plugin)
+        - [Run plugin](#run-plugin)
+    - [Automated testing](#automated-testing)
+        - [Unit testing and code quality](#unit-testing-and-code-quality)
+        - [End-to-end testing (E2E)](#end-to-end-testing-e2e)
+            - [QEMU + Kubernetes](#qemu--kubernetes)
+            - [Starting and stopping a test cluster](#starting-and-stopping-a-test-cluster)
+            - [Running commands on test cluster nodes over ssh](#running-commands-on-test-cluster-nodes-over-ssh)
+            - [Running E2E tests](#running-e2e-tests)
+    - [Communication and contribution](#communication-and-contribution)
+
 
 ## About
 
 ---
-*Note: This is Alpha code and not production ready.*
+ *Note: This is Alpha code and not production ready.*
```

Comment:
This additional space intentional ?

Reply:
Yes, it was treated like a section without that.